### PR TITLE
[6.3] FIX test_positive_create_with_multiple_entities

### PR DESCRIPTION
**Pull request by ldjebran**
The same failure was as described in this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1406907

was affected also by bugs:
https://bugzilla.redhat.com/show_bug.cgi?id=1313056
https://bugzilla.redhat.com/show_bug.cgi?id=1395254

```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_hostgroup.py -v -k "test_positive_create_with_multiple_entities"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 30 items 
2017-06-22 09:26:02 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1199150', '1204686', '1267224', '1221971', '1103157', '1230902', '1214312', '1079482']

2017-06-22 09:26:02 - conftest - DEBUG - Collected 30 test cases


tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities <- robottelo/decorators/__init__.py PASSED

================================================= 29 tests deselected ==================================================
====================================== 1 passed, 29 deselected in 262.54 seconds =======================================
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ make can-i-push?
```

### CODE REVIEW

**Comment by ldjebran** [comment]
update with one test fix and added an option to test_positive_create_with_multiple_entities 
to cover bug https://bugzilla.redhat.com/show_bug.cgi?id=1313056
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_hostgroup.py -v -k "test_positive_create_with_lifecycle_environment or test_positive_create_with_multiple_entities"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 30 items 
2017-06-23 19:09:16 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1199150', '1204686', '1267224', '1221971', '1103157', '1230902', '1214312', '1079482']

2017-06-23 19:09:16 - conftest - DEBUG - Collected 30 test cases


tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_lifecycle_environment <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities <- robottelo/decorators/__init__.py PASSED

================================================= 28 tests deselected ==================================================
====================================== 2 passed, 28 deselected in 304.87 seconds =======================================
```

File: tests/foreman/cli/test_hostgroup.py [hunk 1]
```diff
@@ -318,15 +316,16 @@ def test_positive_create_with_lifecycle_environment(self):
...
+        :BZ: 1359694, 1313056
 
         :CaseImportance: Critical
         """
         org = make_org()
         lc_env = make_lifecycle_environment({'organization-id': org['id']})
         hostgroup = make_hostgroup({
             'lifecycle-environment': lc_env['name'],
-            'organization-id': org['id'],
+            'organization-ids': org['id'],
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: why do we need both `organization-ids` and `query-organization-id`?

**Comment by oshtaier** [comment]
I think to have proper coverage it will be nice to have two cases:
1)
```
make_hostgroup_params = {
             'location-ids': loc['id'],
             'environment-id': env['id'],
             'lifecycle-environment-id': lce['id'],
             'puppet-proxy-id': puppet_proxy['id'],
             'puppet-ca-proxy-id': puppet_proxy['id'],
             'puppet-proxy-id': proxy['id'],
             'puppet-ca-proxy-id': proxy['id'],
             'content-source-id': proxy['id'],
             'content-view-id': cv['id'],
             'domain-id': domain['id'],
             'subnet-id': subnet['id'],
             'organization-ids': org['id'],
             'architecture-id': arch['id'],
             'partition-table-id': ptable['id'],
             'medium-id': media['id'],
             'operatingsystem-id': os['id'],
             'query-organization-id': org['id']
          }
```
+
info and assertion with ids
2) Same, but use names for both create and info procedures

File: tests/foreman/cli/test_hostgroup.py (hunk 1 above)
Comment (ldjebran) [inline tests/foreman/cli/test_hostgroup.py]: as we have 'lifecycle-environment' by name it needs the organization to query/search that name.

**Comment by ldjebran** [comment]
@oshtaier good idea, will investigate if we can have make assertions on info ids when created by ids

**Comment by ldjebran** [comment]
put in do not merge need more investigations

**Comment by ldjebran** [comment]
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_hostgroup.py -v -k "test_positive_create_with_multiple_entities"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 31 items 
2017-06-29 13:09:55 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1199150', '1204686', '1267224', '1221971', '1103157', '1230902', '1214312', '1079482']

2017-06-29 13:09:55 - conftest - DEBUG - Collected 31 test cases


tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities_ids <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities_name <- robottelo/decorators/__init__.py PASSED

================================================= 29 tests deselected ==================================================
====================================== 2 passed, 29 deselected in 238.23 seconds =======================================
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ 
```

**Comment by ldjebran** [comment]
put Do not merge, wait more info from bug reporter about bug resolution https://bugzilla.redhat.com/show_bug.cgi?id=1313056

**Comment by ldjebran** [comment]
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_hostgroup.py -v -k "test_positive_create_with_lifecycle_environment or test_positive_create_with_multiple_entities"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 31 items 
2017-07-11 11:42:14 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1199150', '1204686', '1267224', '1221971', '1103157', '1230902', '1214312', '1079482']

2017-07-11 11:42:14 - conftest - DEBUG - Collected 31 test cases


tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_lifecycle_environment <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities_ids <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities_name <- robottelo/decorators/__init__.py PASSED

================================================= 28 tests deselected ==================================================
====================================== 3 passed, 28 deselected in 257.11 seconds =======================================
```

**Comment by ldjebran** [comment]
@oshtaier @rplevka all comments addressed

**Comment by ldjebran** [comment]
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_hostgroup.py -v -k "test_positive_create_with_lifecycle_environment or test_positive_create_with_multiple_entities"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 31 items 
2017-07-12 12:54:36 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1199150', '1204686', '1267224', '1221971', '1103157', '1230902', '1214312', '1079482']

2017-07-12 12:54:36 - conftest - DEBUG - Collected 31 test cases


tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_lifecycle_environment <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities_ids <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_hostgroup.py::HostGroupTestCase::test_positive_create_with_multiple_entities_name <- robottelo/decorators/__init__.py PASSED

================================================= 28 tests deselected ==================================================
====================================== 3 passed, 28 deselected in 258.36 seconds =======================================
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ 
```

**Pull request merged**